src/dc_receive_imf.rs

```diff
@@ -1602,6 +1601,17 @@ async fn create_or_lookup_group(
         //    .add_protection_msg(context, ProtectionStatus::Protected, false, 0)
         //    .await?;
         //}
+    } else if let Some(chat_id) = chat_id {
+        if create_protected == ProtectionStatus::Protected {
+            let chat = Chat::load_from_db(context, chat_id).await?;
+            if !chat.is_protected() {
+                chat_id
+                    .inner_set_protection(context, ProtectionStatus::Protected)
+                    .await?;
+                recreate_member_list = true; // TODO: is that needed?
+                                             // TODO: groupname / avatar may also need update
```

Comment:
the avatar is used as it comes in, not sure about groupname (if it differs from the one in the qr code)